Added ability to set the keystore\truststore type

habelson opened the pull request:
#436 Added ability to set the keystore\truststore type using either properties or setters
[AMQP-626](https://jira.spring.io/browse/AMQP-626)

pivotal-issuemaster commented:
@habelson Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-amqp&pullRequestId=437)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-projects/spring-amqp&pullRequestId=437) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/faq) for frequently asked questions.

habelson commented:
Just a note, I didn't know what version to use for the @since markings on the new methods, so I left them as "?" thinking someone who knew better could update them.

Additional note, I made sure to default the values to the previous hardcoded values so that no existing code is broken.

artembilan commented:
You can find the version in the JIRA.
So, it is `1.6.2` for you here.

Thank you for contribution!
I'll take a look today, but would like to see some test-cases as well on the matter.
Just in case to cover as much code as possible.

pivotal-issuemaster commented:
@habelson Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-amqp&pullRequestId=437)!

habelson commented:
Understood, I will work on some test cases a bit later today

artembilan commented:
M-m-m. Did you make some reformatting in that class?
Did you change our tabs indents to whitespaces?

habelson commented:
@artembilan Not that I'm aware of. the only lines I changed were the ones I added. They passed the style checks locally. Possible eclipse made changes on its own?

artembilan commented:
> Possible eclipse made changes on its own?

Looks like. Or your Git client.
Try to figure out that issue and revert to tabs, please.

artembilan commented:
Great! I see you have fixed an issue with formatting.
Now you even can add your name to the `@author` list.
But some test is an obligation :smile:

habelson commented:
@artembilan I think it should be good to go now

artembilan commented:
@habelson , looks like not yet.
See changes here on GH.
I think you have a problem with line separator. Or anything else, because we have a changes problem even in the `ssl.properties`.

Check, please, and let us know what's going on with your settings.

artembilan commented on spring-rabbit/src/test/java/org/springframework/amqp/rabbit/connection/SSLConnectionTests.java:
```diff
@@ -1,82 +1,133 @@
+		fb.afterPropertiesSet();
+		try {
+			fb.setUpSSL();
+			// Here we make sure the exception is thrown because setUpSSL() will fail. But we only care about having it load the props
```
Consider, please, do not excess 120 line size.

artembilan commented on spring-rabbit/src/test/java/org/springframework/amqp/rabbit/connection/SSLConnectionTests.java:
```diff
@@ -1,82 +1,133 @@
+
+/**
+ * @author Gary Russell
+ * @since 1.4.4
```
Your name here as well.

artembilan commented on spring-rabbit/src/test/resources/ssl.properties:
```diff
@@ -2,3 +2,5 @@ keyStore=file:/path/to/client/keycert.p12
 keyStore.passPhrase=secret
 trustStore.passPhrase=secret
+keyStore.type=foo
+trustStore.type=bar
```
This is one more annoying formatting issue.
You can configure your IDE to fix that on save.

But right now it is fine and I'm merging your PR after all that formatting hacks from your side :smile: 

Anyway you still owe a description what was wrong with your formatting settings :wink:

habelson commented:
I still do not know what is wrong with my formatting settings. I imported the xml file into eclipse as per instructions. The only way I was able to correct the issue was to revert back to an old version of the file and retype the changes.

My first thought, when it happened the first time, is that maybe my git client did something weird when I stashed my changes to move it to a branch. But I did not do any stashing when I made the test changes and it still happened. So it is still somewhat of a mystery. It seems to only occur to files I've touched, as other files in the project diff fine with what is on github.

artembilan commented:
Merged as https://github.com/spring-projects/spring-amqp/commit/208775dd2e9cb0405d2aa6e6331ff19365ce38c2

Thank you for the contribution.
Looking forward for more!
